pydantic_ai_slim/pydantic_ai/models/bedrock.py

```diff
@@ -408,10 +410,18 @@ async def _messages_create(
             if prompt_variables := model_settings.get('bedrock_prompt_variables', None):
                 params['promptVariables'] = prompt_variables
 
-        if stream:
-            model_response = await anyio.to_thread.run_sync(functools.partial(self.client.converse_stream, **params))
-        else:
-            model_response = await anyio.to_thread.run_sync(functools.partial(self.client.converse, **params))
+        try:
+            if stream:
+                model_response = await anyio.to_thread.run_sync(
+                    functools.partial(self.client.converse_stream, **params)
+                )
+            else:
+                model_response = await anyio.to_thread.run_sync(functools.partial(self.client.converse, **params))
+        except ClientError as e:
+            status_code = e.response.get('ResponseMetadata', {}).get('HTTPStatusCode', 500)
+            raise ModelHTTPError(
+                status_code=status_code, model_name=self.model_name, body=e.response.get('Error')
```

Comment:
I'm not seeing the `Error` key on the `ConverseResponseTypeDef`, so let's use the entire `e.response` as `body`, that also matches the intent of that field better.